File: olymp/city_2015/src/linear.py
MAX_PRIME = 33347


def is_prime(x):
    i = 2
    while i * i <= x:
        if x % i == 0:
            return 0
        i += 1
    return 1


primes = []
for j in xrange(2, MAX_PRIME + 1):
        if is_prime(j):
            primes += [j]
k = int(raw_input())
print primes[primes[k - 1] - 1]

File: olymp/city_2015/src/Linear.java
import java.util.ArrayList;
import java.util.List;

/**
 * Created by lionell on 13.12.2015.
 *
 * @author Ruslan Sakevych
 */
public class Linear {
    public static final int MAX_PRIME = 44611; // 33347
    private boolean[] isPrimeArray = new boolean[MAX_PRIME + 1];
    private List<Integer> primes = new ArrayList<>();

    public Linear() {
        init();
    }

    private void init() {
        eratosthenes();
        for (int i = 2; i <= MAX_PRIME; i++) {
            if (isPrimeArray[i]) {
                primes.add(i);
            }
        }
    }

    public int run(int x) {
        return primes.get(primes.get(x - 1) - 1);
    }

    private void eratosthenes() {
        for (int i = 0; i <= MAX_PRIME; i++) {
            isPrimeArray[i] = true;
        }
        for (int i = 2; i <= MAX_PRIME; i++) {
            if (isPrimeArray[i]) {
                for (int j = i * i; j <= MAX_PRIME; j += i) {
                    isPrimeArray[j] = false;
                }
            }
        }
    }
}
